python/summarize_sgfs.py

```diff
@@ -0,0 +1,120 @@
+from sgfmill import sgf
+import elo
+import os
+import argparse
+
+class SummarizeSGFs:
+  """Summarize Go games in sgfs file format. Each row of sgfs game is a single game in a sgf string."""
+
+  def __init__(
+      self,
+      file_path
+  ):
+    self.file_path = file_path
+    self.sgf_files = []
+    self.players = {}
+    self.results = []
+    self.get_sgfs_files()
+    self.parse_all_sgfs_files()
+
+  def get_sgfs_files(self):
+    """Load all the sgfs file names in file_path directory into a list"""
+    if not os.path.exists(self.file_path):
+      print(f"There is no directory under name {self.file_path}")
+      return None
+    self.sgf_files = os.listdir(self.file_path)
+
+    for file in self.sgf_files:
+      if file.split(".")[-1].lower() != "sgfs":
+        self.sgf_files.remove(file)
+    print(f"Total {len(self.sgf_files)} sgfs files have been found under directory {self.file_path}.")
+
+
+  def parse_all_sgfs_files(self):
+    """Parse all sgfs files"""
+    idx=1
+    for file in self.sgf_files:
+        self.parse_one_sgfs_file(file)
+        print(f"Parsed {idx} files out of {len(self.sgf_files)} files.")
+        idx+=1
+    self.estimate_elo()
+
+
+  def estimate_elo(self):
+    """Estimate and print elo values. This function must be called after parse all the sgfs files"""
+    data = []
+    pla_names = list(self.players.keys())
+    n_pla = len(pla_names)
+    for i in range(n_pla-1):
+      for j in range(i+1,n_pla):
+        won = self.results[self.players[pla_names[i]]][self.players[pla_names[j]]]
+        lost = self.results[self.players[pla_names[j]]][self.players[pla_names[i]]]
+        total_games = won + lost
+        data.extend(elo.likelihood_of_games(pla_names[i], pla_names[j], total_games, won/total_games, False))
+
+    data.extend(elo.make_center_elos_prior(pla_names, 0))
```

Comment:
Based on this suggestion, we have made the prior input in constructor, and it can also input from command line.  We also add the print Elo prior before printing the result.